### PR TITLE
feat(helm): Add ingress to the helm chart

### PR DESCRIPTION
**Pull request by Avihais12344**
Hello,

I have added an ingress to the helm chart, making the exporter accessiable to the outside world.

It's a simple, but affective change.

### CODE REVIEW

**Comment by burningalchemist** [comment]
Hey @Avihais12344, thanks for your contribution. 

In general, the idea of exposing exporters to the public internet is quite insecure, especially in case of sql_exporter. sql_exporter usually has direct access to the database instances, and potentially can be used for (D)DOS attacks. Also, I'd not enable ingress by default.

It is possible to reduce the public access by configuring TLS certificates and/or Basic authentication as per [this document](https://github.com/prometheus/exporter-toolkit/blob/master/docs/web-configuration.md). So if I'd expose sql_exporter to the public, it'd be in combination with the configuration above.

Could you please explain the rationale behind having the public access to the sql_exporter or some use cases? This topic didn't pop up so far.

Hey @allanger, could you please review the PR whenever/if you have some free time? I'm not convinced so far it's a good idea.

**Comment by allanger** [comment]
I'm not sure whether it's a good idea, too. I would suggest having an agent inside the same cluster where the operator is deployed that is either sending data to the central place, or that is being scraped. 

But if the ingress is a must-have, I'm not sure how it would handle the TLS auth. I assume you would expect to receive encrypted traffic, but in most cases ingress terminates TLS and after leaving the ingress pod the traffic is not encrypted, at least, that's how it's usually configured in my experience. The same can apply to the Basic Auth, but it can't be enforced on the chart level, because different ingress controllers are handling it differently.

Regarding the PR itself, I find the TLS definition a bit confusing, but it seems to be doing what it's intended to do, and I'm not sure whether I can suggest a better solution atm. 
I would not add a helper for the tls, and would try to separate the secret creation from the ingress, but then it would become less dynamic

**Comment by Avihais12344** [comment]
Hello @burningalchemist and @allanger. I understand your consern, but in my case there's a lot of reason to do that.
First of all, my orginization works inside a closed network (orginization network), where the prometheus that scrapes me doesn't sits in my `K8S` cluster.
Also, we can't install the `ServiceMonitor` CRD, becasue it isn't installed in our managed K8S environment, and we can't install it there. Because we are in a closed network, we don't afraid of `DDos` attacks. 

Even for the outside, free, world we can handle auth and TLS pretty easily. TLS we configure at the ingress, and for the auth, we can use Ingress Controllers, like the [nginx](https://kubernetes.github.io/ingress-nginx/examples/auth/basic/) one to handle authentication.

I understand your consern, so I also added to the readme a warning about it.

File: helm/templates/secret.tls.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# ---------------------------------------------------------------------
+# -- This secret holds the config file of sql_exporter
```
Comment (allanger) [inline helm/templates/secret.tls.yaml:2]: It doesn't hold the config, does it?

**Comment by allanger** [comment]
And you also need to bump the chart version

File: helm/README.md [hunk 2]
```diff
@@ -23,6 +28,15 @@ helm repo add sql_exporter https://burningalchemist.github.io/sql_exporter/
 helm install sql_exporter/sql-exporter
 ```
 
+### Warning on enabling ingress
+
+Please enable the ingress with caution.
```
[the remark this answers is not in the document]
Comment (Avihais12344) [reply helm/README.md]: Fixed!

File: helm/templates/secret.tls.yaml (hunk 1 above)
Comment (Avihais12344) [reply helm/templates/secret.tls.yaml]: You are right, fixed!

**Comment by Avihais12344** [comment]
> And you also need to bump the chart version

You are right, saw it at the CI, fixed it!

Thank you for your CR!

**Comment by Avihais12344** [comment]
> Hey @Avihais12344, just double checking that you didn't update the docs manually, here is the command to do some automation (it should also help with unexpected changes in a new PR next time): https://github.com/burningalchemist/sql_exporter/tree/master/helm#dev-notes

No, I saw the note before, I have updated the docs using the command `make gen_docs`.

**Comment by burningalchemist** [comment]
Awesome, I'm just checking. 👍

**Review by allanger — APPROVED** [review]

**Pull request merged**